torchok/models/heads/detection/detr.py

```diff
@@ -0,0 +1,377 @@
+                with normalized coordinate format (cx, cy, w, h) and shape [nb_dec, bs, num_query, 4].
+            gt_bboxes: Ground truth bboxes for each image with shape (num_gts, 4) in [tl_x, tl_y, br_x, br_y] format.
+            gt_labels: Ground truth class indices for each image with shape (num_gts, ).
+            img_metas: List of image meta information.
+
+        Returns:
+            dict[str, Tensor]: A dictionary of loss components.
+        """
+        global_total_loss = 0
+        tagged_loss_dict = defaultdict(int)
+        for cls_scores, bbox_preds in zip(all_cls_scores, all_bbox_preds):
+            num_imgs = cls_scores.size(0)
+            cls_scores_list = [cls_scores[i] for i in range(num_imgs)]
+            bbox_preds_list = [bbox_preds[i] for i in range(num_imgs)]
+            cls_reg_targets = self.get_targets(cls_scores_list, bbox_preds_list,
+                                               gt_bboxes, gt_labels, img_metas)
+            (labels_list, label_weights_list, bbox_targets_list, bbox_weights_list,
+             num_total_pos, num_total_neg) = cls_reg_targets
+            labels = torch.cat(labels_list, 0)
+            label_weights = torch.cat(label_weights_list, 0)
+            bbox_targets = torch.cat(bbox_targets_list, 0)
+            bbox_weights = torch.cat(bbox_weights_list, 0)
+
+            # classification loss
+            cls_scores = cls_scores.reshape(-1, self.cls_out_channels)
+            # construct weighted avg_factor to match with the official DETR repo
+            cls_avg_factor = num_total_pos * 1.0 + num_total_neg * self.bg_cls_weight
+            if self.sync_cls_avg_factor:
+                cls_avg_factor = reduce_mean(torch.tensor([cls_avg_factor], device=cls_scores.device))
+            cls_avg_factor = max(cls_avg_factor, 1)
+
+            # Compute the average number of gt boxes across all gpus, for
+            # normalization purposes
+            num_total_pos = torch.clamp(torch.tensor(reduce_mean(num_total_pos)), min=1).item()
+
+            # construct factors used for rescale bboxes
+            factors = []
+            for bbox_pred, img_meta in zip(bbox_preds, img_metas):
+                img_h, img_w, _ = img_meta['img_shape']
+                factor = bbox_pred.new_tensor([img_w, img_h, img_w, img_h]).unsqueeze(0).repeat(bbox_pred.size(0), 1)
+                factors.append(factor)
+            factors = torch.cat(factors, 0)
+
+            # DETR regress the relative position of boxes (cxcywh) in the image,
+            # thus the learning target is normalized by the image size. So here
+            # we need to re-scale them for calculating IoU loss
+            bbox_preds = bbox_preds.reshape(-1, 4)
+            bboxes = bbox_cxcywh_to_xyxy(bbox_preds) * factors
+            bboxes_gt = bbox_cxcywh_to_xyxy(bbox_targets) * factors
+
+            total_loss, tagged_loss_values = joint_loss(
```

Comment:
Did you try to return list of all staff that need joint loss to task and compute loss in task? Or it's bad idea because we return list of tensors and it can increase total memory? Or explain reason that it is bad to do it.